### PR TITLE
Add functional tests for RazorPage routes

### LINKED ISSUE #5842: Add Functional Tests: Routing

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #5842.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 1]
```diff
@@ -167,6 +167,19 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
         }
 
         [Fact]
+        public async Task Page_Route_RouteDataCoercion()
```
Comment (Eilon) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Not sure what this test method name means. What gets coerced with what? I prefer when test names are fairly specific about what they expect...

**Comment by ryanbrandenburg** [comment]
:up::date: how's that name look?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 2]
```diff
@@ -167,6 +166,19 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
         }
 
         [Fact]
+        public async Task Page_Route_RouteData_StringToIntCoercion_NotFound()
```
Comment (Eilon) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Sorry I still don't get what this is testing 😢 🐼 

Test names usually follow the pattern `SuchAndSuchFeature_InSuchAndSuchConditions_ExpectsSuchAndSuchResult`.

So, what component is being tested? Under what interesting setup/condition? And what is the expected result?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 3]
```diff
@@ -167,6 +166,19 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
...
+        {
+            // Arrange
+            var routeRequest = new HttpRequestMessage(HttpMethod.Get, "http://localhost/RouteData/pizza");
+           
+            // Act
+            var routeResponse = await Client.SendAsync(routeRequest);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, routeResponse.StatusCode);
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Is there a test already that tests the success case for this scenario?

Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: There's one right below this now.

**Comment by ryanbrandenburg** [comment]
:up::date: @Eilon hopefully this is named informativly enough?

**Pull request closed without merging**